sdk/core/Azure.Core/api/Azure.Core.net461.cs

```diff
@@ -211,6 +225,8 @@ public partial class RequestFailedException : System.Exception, System.Runtime.S
         public RequestFailedException(string message, System.Exception? innerException) { }
         public string? ErrorCode { get { throw null; } }
         public int Status { get { throw null; } }
+        public static Azure.RequestFailedException Create(Azure.Response response, Azure.ResponseError? error = null, System.Exception? innerException = null) { throw null; }
+        public static System.Threading.Tasks.ValueTask<Azure.RequestFailedException> CreateAsync(Azure.Response response, Azure.ResponseError? error = null, System.Exception? innerException = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
```

Comment:
Curious: did we consider an approach where we take the custom parser directly into the constructor/factory method instead of adding it to pipeline build options?

Reply:
> What is the advantage of having it baked into the pipeline?

So we can get the parser set on the Response (currently it's done by the `HttpPipelineTransportPolicy` just like the Sanitizer)

If we don't set it on the `Response` we have to solve the problem of how to get the parser to the factory method.

Reply:
> If we don't set it on the Response we have to solve the problem of how to get the parser to the factory method.

Would passing the parser as a parameter into the factory method not work to get the parser to the factory method?

Reply:
One of the reasons we stashed it on the `Response` in the first place was to avoid having to pass the parser around. 

Could you say more about the concern with having it be part of the pipeline build?

Reply:
I also like the idea of passing this into the factory method. This is where the parser will be used so it seems reasonable to pass it if you want to do something custom.

Reply:
I was just wondering if we'd considered that approach.  We're adding a lot of types and constructors to support putting it on Response (which we'll probably need in the future), and if the RequestFailedException constructor is always called from a client method implementation, it might reduce complexity to just pass it in at the call site when the exception is created.  If there are other places we create exceptions, then I agree this may not be ideal.

Reply:
> I was just wondering if we'd considered that approach. We're adding a lot of types and constructors to support putting it on Response (which we'll probably need in the future), and if the RequestFailedException constructor is always called from a client method implementation, it might reduce complexity to just pass it in at the call site when the exception is created. If there are other places we create exceptions, then I agree this may not be ideal.

Yes, in fact that was the original design. But in the arch board discussion on this topic, the idea was raised to stash it on the `Response` instead. After implementing it this way, I like the idea even more. It avoids the caller from having to make decisions when raising an exception, just once when configuring the pipeline. In addition, to your point, the context for how to raise an exception is always with `Response`. The advantage there is this avoids ever having to refactor just to make the error customization available in additional code paths. 

That said, I'm happy to revisit the decision if we agree that the advantages are more compelling.